tempto-kafka/src/main/java/io/prestodb/tempto/fulfillment/table/kafka/KafkaMessage.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prestodb.tempto.fulfillment.table.kafka;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static java.util.Objects.requireNonNull;
+
+public class KafkaMessage
+{
+    private final OptionalInt partition;
+    private final Optional<byte[]> key;
+    private final byte[] value;
+
+    public KafkaMessage(byte[] value, OptionalInt partition) {
+        this(Optional.empty(), value, partition);
+    }
+
+    public KafkaMessage(byte[] value) {
+        this(Optional.empty(), value, OptionalInt.empty());
+    }
+
+    public KafkaMessage(byte[] key, byte[] value) {
+        this(Optional.of(key), value, OptionalInt.empty());
+    }
+
+    public KafkaMessage(Optional<byte[]> key, byte[] value, OptionalInt partition)
+    {
+        this.partition = requireNonNull(partition, "partition is null");
+        this.key = requireNonNull(key, "key is null");
```

Comment:
Can be empty? Is there a difference between `key` being `Optional.of(new byte[0])` and `Optional.empty()`?

Reply:
The Kafka client API treats no-key scenario in special way. You pass null instead empty data.